minari/dataset/minari_storage.py

```diff
@@ -1,25 +1,31 @@
 from __future__ import annotations
 
+import json
 import os
 import pathlib
-from collections import OrderedDict
-from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, Union
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Dict, Iterable, List, Optional, Union
 
 import gymnasium as gym
-import h5py
 import numpy as np
 from gymnasium.envs.registration import EnvSpec
 
 from minari.serialization import deserialize_space, serialize_space
 
 
 PathLike = Union[str, os.PathLike]
+_METADATA_FILE_NAME = "metadata.json"
 
 
-class MinariStorage:
+class MinariStorage(ABC):
     """Class that handles disk access to the data."""
 
-    def __init__(self, data_path: PathLike):
+    def __init__(
+        self,
+        data_path: pathlib.Path,
```

Comment:
The reason for this is that I often need to use joinpath and check existence, so it is easier if data_path is already a pathlib.Path. So I do the casting in `.read` method and I thought to do it once and for all(?).
This can be changed of course, no strong opinion on that.